File: src/main/java/com/github/zxh/classpy/classfile/attribute/SourceFileAttribute.java
package com.github.zxh.classpy.classfile.attribute;

import com.github.zxh.classpy.classfile.reader.ClassReader;
import com.github.zxh.classpy.classfile.datatype.U2CpIndex;

/*
SourceFile_attribute {
    u2 attribute_name_index;
    u4 attribute_length;
    u2 sourcefile_index;
}
 */
public class SourceFileAttribute extends AttributeInfo {

    private U2CpIndex sourceFileIndex;
    
    @Override
    protected void readInfo(ClassReader reader) {
        sourceFileIndex = reader.readU2CpIndex();
    }
    
}

File: src/main/kotlin/com/github/zxh/classpy/classfile/attribute/SourceFileAttribute.kt
package com.github.zxh.classpy.classfile.attribute;

/*
SourceFile_attribute {
    u2 attribute_name_index;
    u4 attribute_length;
    u2 sourcefile_index;
}
 */
class SourceFileAttribute : AttributeInfo() {

    init {
        u2cp("source_file_index");
    }

}
